### PR TITLE
Upgraded Condition Resource

### PR DESCRIPTION
**Pull request by yashrajbothra**
#### Short description of what this resolves:
Upgraded FHIR Condition Resource

#### Changes proposed in this pull request:

-
-
-

### CODE REVIEW

**Comment by bradymiller** [comment]
@yashrajbothra , just let me know when this is ready for a review :)

**Comment by yashrajbothra** [comment]
@bradymiller Sure :)

**Comment by yashrajbothra** [comment]
@bradymiller All Set to Review :)

File: interface/patient_file/summary/stats_full.php [hunk 1]
```diff
@@ -181,6 +181,8 @@ function newEncounter() {
                 <th scope="col"><?php echo xlt('Occurrence'); ?></th>
                 <?php if ($focustype == "allergy") { ?>
                   <th scope="col"><?php echo xlt('Reaction'); ?></th>
+                <?php }
+                if ($focustype == "allergy" || $focustype == "medical_problem") { ?>
```
Comment (bradymiller) [inline interface/patient_file/summary/stats_full.php:185]: note this didn't work well previously (it didn't work for edit ui of items), so instead leveraged the javascript method:
https://github.com/openemr/openemr/commit/3e1e77b1b0e1f87d030a607713974e4d8c6aaa70

Also, the medical_problem should use it's own list, which will default to Confirmed

File: src/Services/AllergyIntoleranceService.php [hunk 2]
```diff
@@ -115,7 +113,7 @@ public function getAll($search = array(), $isAndCondition = true)
                 UuidRegistry::uuidToString($row['practitioner']) :
                 $row['practitioner'];
             if ($row['diagnosis'] != "") {
-                $this->addDiagnosis($row['diagnosis']);
+                $row['diagnosis'] = $this->addDiagnosis($row['diagnosis']);
```
Comment (bradymiller) [inline src/Services/AllergyIntoleranceService.php:116]: it looks like you removed the addDiagnosis function below?

Comment (bradymiller) [reply src/Services/AllergyIntoleranceService.php:116]: disregard above comment, i see what you did (now in base class) :)

File: src/Services/ProcedureService.php [hunk 3]
```diff
@@ -0,0 +1,184 @@
+<?php
+
```
Comment (bradymiller) [inline src/Services/ProcedureService.php:2]: Is this ProcedureService supposed to be here?

**Comment by bradymiller** [comment]
review complete :)

File: src/Services/ProcedureService.php (hunk 3 above)
Comment (yashrajbothra) [reply src/Services/ProcedureService.php:2]: Oops It was WIP for Our Next Resource :sweat_smile:

File: interface/patient_file/summary/stats_full.php (hunk 1 above)
Comment (yashrajbothra) [reply interface/patient_file/summary/stats_full.php:185]: Did you mean Default in Database or Default in UI?

Comment (yashrajbothra) [reply interface/patient_file/summary/stats_full.php:185]: Also, this file will not be used in edit UI so will this change have any effect here?

Comment (bradymiller) [reply interface/patient_file/summary/stats_full.php:185]: Regarding default, meant in the UI.

You are correct that I was mistaken on this file (assumed it was the add edit script), so disregard that part of the comment :)

File: sql/5_0_2-to-6_0_0_upgrade.sql [hunk 4]
```diff
@@ -1921,3 +1921,14 @@ INSERT INTO list_options(list_id,option_id,title,seq) VALUES ('allergyintoleranc
 #IfMissingColumn ar_activity deleted
 ALTER TABLE `ar_activity` ADD COLUMN `deleted` datetime DEFAULT NULL COMMENT 'NULL if active, otherwise when voided';
```
Comment (bradymiller) [inline sql/5_0_2-to-6_0_0_upgrade.sql:1922]: these changes also need to go into database.sql

**Comment by yashrajbothra** [comment]
Couldn't help travis this time :(

**Comment by bradymiller** [comment]
hi @yashrajbothra , Code looks good and bringing into codebase (I'll beat up travis for you :) ) :
tenor (19)

Thanks for the improvements! -brady

**Pull request merged**